amethyst_assets/src/formats.rs

```diff
@@ -32,6 +32,11 @@ where
     }
 }
 
+impl<D: FormatRegisteredData> SerializableFormat<D> for RonFormat where
```

Comment:
This seems to be missing the registration macro, which reminds me of: Please add tests checking that `RonFormat` and `JsonFormat` are properly registered.

Reply:
@torkleyy 
I'm not sure we can register format for `RonFormat` and `JsonFormat`, as their underlying data parameter (`FormatRegisteredData`) is generic.
I think it'll be possible to register `RonFormat` and `JsonFormat` only if we introduce a new format type for them. (Am I right?..) And with this change we can also introduce a new kind of asset, though maybe it goes out of scope of this PR...

For now I've just dropped `SerializableFormat` implementation for both of them, as far as I notice we don't even use it anywhere for these types. I believe we can bring it back once we decide to introduce something like a `ConfigFile` asset type and the corresponding data type.

Let me know if that sounds reasonable.

Reply:
Generic formats weren't really meant for registration. Every type has it's own storage, and here the type is abstract. Rust prevents generic types from having a static member, so in current form it's impossible.